sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ObservableEventBatch.md

```diff
@@ -0,0 +1,152 @@
+# Observable Event Data Batch
+
+This sample demonstrates how to write an `ObservableEventDataBatch` class that wraps an `EventDataBatch` in order to allow an application to read events that have been added to a batch. This is unlike the standard `EventDataBatch` class which restricts the application from accessing `EventData` instances once they have been accepted into the batch. `EventDataBatch` has this limitation to ensure that the state of the batch remains consistent and valid as events are added, and that it can successfully be published.
+
+## Considerations
+
+While the `ObservableDataBatch` may seem desirable, there are several nuances that should be considered before using it in your application.   In order to make sure that once an event is successfully added to the batch it will be sent, the `EventDataBatch` creates a hidden copy of the events which it uses to publish to the Event Hub. Without this, the application could make changes to them, and potentially invalidate the batch size calculations,  making it too large to publish.   If `EventDataBatch` held onto the original event after copying and made them accessible, applications altering the events would not be altering the event that was published, and the events that it sees as belonging to the batch would no longer match the actual batch content that will be published. 
+
+
+Another issue is with equality. `EventData` objects do not have a strong and deterministic way to define equality, since the meaning of two events being equal can be different depending on the application. This creates confusion and can cause unnecessary issues in some applications.
+
+Using this approach **does not** protect from the inherent risks as discussed above, and is not recommended unless the appplication is willing to assume those risks. If the `EventData` objects exposed through the `Events` variable are altered, these changes will not be reflected in the events held in the actual Batch. 
+
+#  Implicit Casting
+
+The most straightforward way to implement this functionality is by creating a class that wraps a normal `EventDataBatch` instance, as well as a read-only list of the events that have been successfully added to the batch. Even though the list itself is readonly, the `EventData` objects are not, and if they are changed after the object has been added to the batch, this will NOT be reflected in the actual data stored by the `EventDataBatch` itself. 
```

Comment:
```suggestion
The most straightforward way to implement this functionality is by creating a class that wraps a normal `EventDataBatch` instance, as well as a read-only list of the events that have been successfully added to the batch. Even though the list itself is read-only, the `EventData` objects are mutable, and if they are changed after the object has been added to the batch, this will **not** be reflected in the actual data stored by the `EventDataBatch` itself. 
```